### PR TITLE
fetchLines() in bom class : fetch child bom

### PR DESCRIPTION
**Pull request by atm-lena**
# FIX|Fix #
We need to fetch bom child of the bom line if it exists when we use "fetchlines()" for a BOM.

### CODE REVIEW

File: htdocs/bom/class/bom.class.php [hunk 1]
```diff
@@ -396,6 +396,16 @@ public function fetchLines()
 		$this->lines = array();
 
 		$result = $this->fetchLinesCommon();
+
+		//associate to each line childbom if exist
+		if (!empty($this->lines)) {
+			foreach ($this->lines as $line) {
+				$bom_child = new BOM($this->db);
+				$res = $bom_child->fetch($line->fk_bom_child);
```
Comment (eldy) [inline htdocs/bom/class/bom.class.php:404]: fetch and the init of new BOM must be done only if  $line->fk_bom_child is set

Comment (atm-lena) [reply htdocs/bom/class/bom.class.php]: @eldy Done

File: htdocs/bom/class/bom.class.php [hunk 2]
```diff
@@ -396,6 +396,18 @@ public function fetchLines()
 		$this->lines = array();
 
 		$result = $this->fetchLinesCommon();
+
+		//associate to each line childbom if exist
+		if (!empty($this->lines)) {
+			foreach ($this->lines as $line) {
+				if($line->fk_bom_child > 0) {
+					$bom_child = new BOM($this->db);
+					$res = $bom_child->fetch($line->fk_bom_child);
```
Comment (eldy) [inline htdocs/bom/class/bom.class.php:405]: There is a risk of infinite loop if in database a bom A has child B that has child A...
Do you really need have a subrequest for child to load "immediatly" the child.
Can't you load it only when "you need it" ?

**Comment by eldy** [comment]
Closed. Opened for too long. Missing answer.

**Pull request closed without merging**